File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/webdriver/WebDriverResponse.java
package ru.kmorozov.gbd.core.logic.connectors.webdriver;

import org.apache.commons.io.input.CharSequenceInputStream;
import ru.kmorozov.gbd.core.logic.connectors.Response;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.Charset;

class WebDriverResponse implements Response {

    private final String response;

    WebDriverResponse(final String response) {
        this.response = response;
    }


    @Override
    public InputStream getContent() throws IOException {
        return new CharSequenceInputStream(this.response, Charset.forName("UTF-8"));
    }

    @Override
    public String getImageFormat() {
        return null;
    }

    @Override
    public void close() throws IOException {

    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/webdriver/WebDriverResponse.kt
package ru.kmorozov.gbd.core.logic.connectors.webdriver

import ru.kmorozov.gbd.core.logic.connectors.Response
import java.io.ByteArrayInputStream
import java.io.IOException
import java.io.InputStream
import java.nio.charset.StandardCharsets

internal class WebDriverResponse(private val response: String) : Response {


    override val content: InputStream
        @Throws(IOException::class)
        get() = ByteArrayInputStream(response.toByteArray(StandardCharsets.UTF_8));

    override val imageFormat: String
        get() = "unknown"

    @Throws(IOException::class)
    override fun close() {

    }
}
